fastly/resource_fastly_service_waf_configuration.go

```diff
@@ -273,30 +276,37 @@ func resourceServiceWAFConfigurationV1Update(d *schema.ResourceData, meta interf
 		MinTimeout: WAFStatusCheckMinTimeout,
 		Check:      DefaultWAFDeploymentChecker(conn),
 	}
-	err = statusCheck.waitForDeployment(wafID, latestVersion)
+	err = statusCheck.waitForDeployment(ctx, wafID, latestVersion)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-	return resourceServiceWAFConfigurationV1Read(d, meta)
+	return resourceServiceWAFConfigurationV1Read(ctx, d, meta)
 }
 
-func resourceServiceWAFConfigurationV1Read(d *schema.ResourceData, meta interface{}) error {
+func resourceServiceWAFConfigurationV1Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 
 	latestVersion, err := getLatestVersion(d, meta)
-	if errRes, ok := err.(*gofastly.HTTPError); ok {
-		if errRes.StatusCode == 404 {
-			log.Printf("[DEBUG] WAF (%s) was not found - removing from state", d.Get("waf_id").(string))
-			d.SetId("")
-			return nil
+	if err != nil {
+		if errRes, ok := err.(*gofastly.HTTPError); ok {
+			if errRes.StatusCode == 404 {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity:      diag.Warning,
+						Summary:       fmt.Sprintf("WAF (%s) not found - removing from state", d.Get("waf_id")),
+						AttributePath: cty.Path{cty.GetAttrStep{Name: "waf_id"}},
```

Comment:
I looked up the cty dependency and to be honest I don't understand it. Maybe you can help explain its purpose to me.

Reply:
My understanding of the cty dependency is that it is the building blocks to represent the schema and in this case it is a means to help direct a user to the location in the HCL where there is a problem.  In this case it would direct to the waf_id.